src/server_manager/electron_app/index.ts

```diff
@@ -149,21 +149,21 @@ function workaroundDigitalOceanApiCors() {
   const headersFilter = {urls: ['https://api.digitalocean.com/*']};
   electron.session.defaultSession.webRequest.onHeadersReceived(
       // tslint:disable-next-line:no-any
-      headersFilter, (details: any, callback: Function) => {
+      headersFilter, (details: electron.OnHeadersReceivedListenerDetails, callback: (response: any) => void) => {
+        console.error(details);
         if (details.method === 'OPTIONS') {
-          details.responseHeaders['access-control-allow-origin'] = ['outline://web_app'];
+          details.responseHeaders['access-control-allow-origin'] = 'outline://web_app';
           if (details.statusCode === 403) {
             details.statusCode = 200;
             details.statusLine = 'HTTP/1.1 200';
-            details.responseHeaders['status'] = ['200'];
-            details.responseHeaders['access-control-allow-headers'] =
-                [details.headers['Access-Control-Request-Headers']];
-            details.responseHeaders['access-control-allow-credentials'] = ['true'];
+            details.responseHeaders['status'] = '200';
+            details.responseHeaders['access-control-allow-headers'] = 'authorization,content-type';
```

Comment:
Don't these need to match (be a superset of) the request headers?  What was wrong with the previous solution? Hard-coding runs the risk of not keeping up with some changes to the headers that are requested

Reply:
It turns out `details.headers` doesn't exist. That's what's causing the exception. The API has no mention of it: https://github.com/electron/electron/blob/v7.0.1/docs/api/web-request.md#webrequestonheadersreceivedfilter-listener

The logging was a way to verify that. There's no replacement field.

In any case, I'm using '*', which works.

Reply:
Oops. Makes sense!